### PR TITLE
Fix for performance slowdown observed during packaging and versioning…

### PR DESCRIPTION
**Pull request by RohitKandimalla**
… of FHIR measures

## Checklist

- [ ] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [ ] Tests are included and test edge cases.
- [ ] Tests have been run locally and pass.

## Screenshots (if appropriate)

- [ ] None applicable

### CODE REVIEW

File: src/main/java/cqltoelm/parsers/MATCQL2ELMListener.java [hunk 1]
```diff
@@ -414,13 +414,15 @@ private Element resolve(String identifier, TranslatedLibrary library) {
...
             IncludeDef def = (IncludeDef) element;
-            libraries.add(def.getPath() + "-" + def.getVersion() + "|" + def.getLocalIdentifier());
             graph.addEdge(currentContext, def.getPath() + "-" + def.getVersion() + "|" + def.getLocalIdentifier());
             libraryAccessor = def;
-            try {
-                parseChildLibraries(def);
-            } catch (IOException e) {
-                e.printStackTrace();
+            if (!libraries.contains(def.getPath() + "-" + def.getVersion() + "|" + def.getLocalIdentifier())) {
+                libraries.add(def.getPath() + "-" + def.getVersion() + "|" + def.getLocalIdentifier());
```
Comment (jkotanchik-SB) [inline src/main/java/cqltoelm/parsers/MATCQL2ELMListener.java:420]: How about moving the `add` inside the try/catch block, right after `parseChildLibraries` to ensure we're only adding the library if the method is successfully. Though, we don't do anything different in the error case 🤣

File: src/main/java/cqltoelm/parsers/MATCQL2ELMListener.java [hunk 2]
```diff
@@ -414,13 +414,15 @@ private Element resolve(String identifier, TranslatedLibrary library) {
...
+                try {
+                    parseChildLibraries(def);
```
Comment (jkotanchik-SB) [inline src/main/java/cqltoelm/parsers/MATCQL2ELMListener.java:422]:
```suggestion
                try {
                    parseChildLibraries(def);
                    libraries.add(def.getPath() + "-" + def.getVersion() + "|" + def.getLocalIdentifier());
```
How about moving the `add` inside the try/catch block, right after `parseChildLibraries` to ensure we're only adding the library if the method is successfully. Though, we don't do anything different in the error case

File: src/main/java/mat/server/CQLServiceImpl.java [hunk 3]
```diff
@@ -1642,13 +1642,7 @@ public GetUsedCQLArtifactsResult getUsedCQlArtifacts(String xml) {
...
         }
 
-        String cqlString  = CQLUtilityClass.getCqlString(cqlModel, "").getLeft();
-        SaveUpdateCQLResult cqlResult;
-        if (cqlModel.isFhir()) {
-            cqlResult = parseFhirCQLForErrors(cqlModel, cqlString);
-        } else {
-            cqlResult = CQLUtil.parseQDMCQLLibraryForErrors(cqlModel, getCqlLibraryDAO(), exprList);
-        }
+        SaveUpdateCQLResult cqlResult = CQLUtil.parseQDMCQLLibraryForErrors(cqlModel, getCqlLibraryDAO(), exprList);
```
[the remark this answers is not in the document]
Comment (RohitKandimalla) [reply src/main/java/mat/server/CQLServiceImpl.java:1645]: yup no longer needed

File: src/main/java/cqltoelm/parsers/MATCQL2ELMListener.java (hunk 2 above)
Comment (RohitKandimalla) [reply src/main/java/cqltoelm/parsers/MATCQL2ELMListener.java:422]: Yup it doesn't matter

Comment (jkotanchik-SB) [reply src/main/java/cqltoelm/parsers/MATCQL2ELMListener.java:422]: Yes, pls.

**Review by jkotanchik-SB — APPROVED** [review]
